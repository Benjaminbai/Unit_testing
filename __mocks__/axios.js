
const mockAxios = jest.requireActual('Axios')

export default mockAxios;