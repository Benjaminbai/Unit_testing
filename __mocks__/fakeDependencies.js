jest.mock('react-router-dom', () => {
    const originalModule = jest.requireActual('react-router-dom');

    return {
        __esModule: true,
        ...originalModule,
        useLocation: jest.fn().mockReturnValue({
            pathname: '',
            search: '',
            hash: '',
            state: null,
            key: '',
        }),
        useHistory: () => ({
            push: jest.fn(),
            location: {}
        }),
        Redirect: jest.fn().mockReturnValue({}),
    }
})
window.matchMedia = jest.fn().mockImplementation((query) => {
    const obj = {
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(),
        removeListener: jest.fn(),
    };

    return obj;
})

jest.mock('react', () => ({
    ...jest.requireActual('react'),
    useLayoutEffect: jest.requireActual('react').useEffect
}))