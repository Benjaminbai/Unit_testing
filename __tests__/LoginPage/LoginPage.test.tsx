import '../../__mocks__/fakeDependencies'
import React from 'react'
import { mount, render } from 'enzyme'
import LoginPage from '../../pages/LoginPage/LoginPage'
import { BrowserRouter as Router } from 'react-router-dom'

import { render as RTLRender, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import API from '../../api/LoginPage'

describe('<LoginPage />', () => {
    test('expect the LoginPage exist', () => {
        const LoginPageComponent = mount(<Router><LoginPage /></Router>)
        expect(LoginPageComponent.exists()).toBe(true)
    })

    test('expect LoginPage unmount', () => {
        const LoginPageComponent = mount(<Router><LoginPage /></Router>)
        LoginPageComponent.unmount()
        expect(() => {
            LoginPageComponent.instance()
        }).toThrow()
    })

    // 快照
    test('the source code changed or not', () => {
        const LoginPageComponent = render(<Router><LoginPage /></Router>)
        expect(LoginPageComponent).toMatchSnapshot()
    })

    test('userLogin api', () => {
        const spyFn = jest.spyOn(API, 'userLogin')
        RTLRender(<Router><LoginPage /></Router>)
        userEvent.click(screen.getByText(/登录/i))
        expect(spyFn).toBeCalled()
    })
})