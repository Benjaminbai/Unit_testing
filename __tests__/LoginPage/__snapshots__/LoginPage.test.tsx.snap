// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginPage /> the source code changed or not 1`] = `
<div
  class="limiter"
>
  <div
    class="container-login100"
  >
    <div
      class="wrap-login100"
    >
      <div
        class="login100-pic"
      >
        <img
          alt="IMG"
          src="undefined/assets/images/img-01.png"
        />
      </div>
      <div
        class="login100-form"
      >
        <span
          class="login100-form-title"
        >
          Quark
        </span>
        <div
          class="wrap-input100"
        >
          <input
            class="input100"
            name="username"
            placeholder="Username"
            type="text"
            value=""
          />
          <span
            class="focus-input100"
          />
          <span
            class="symbol-input100"
          >
            <i
              aria-hidden="true"
              class="fa fa-user"
            />
          </span>
        </div>
        <div
          class="wrap-input100"
        >
          <input
            class="input100"
            name="pass"
            placeholder="Password"
            type="password"
            value=""
          />
          <span
            class="focus-input100"
          />
          <span
            class="symbol-input100"
          >
            <i
              aria-hidden="true"
              class="fa fa-lock"
            />
          </span>
        </div>
        <div
          class="container-login100-form-btn"
        >
          <button
            class="login100-form-btn"
          >
            登录
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
